common/zeroconf/zeroconf.cpp

```diff
@@ -53,15 +53,34 @@ namespace Aseba
 		else
 		{
 			pending[serviceref] = std::unique_ptr<ZeroconfDiscoveryRequest>(new ZeroconfDiscoveryRequest());;
-			// Responses from the DNS Service will subsequently be handled in the watcher thread,
-			// where they will be dispatched to Zeroconf::cb_Register by DNSServiceProcessResult.
+			// Block until daemon replies. The DiscoveryRequest will be moved from pending
+			// to services by the cb_Register callback.
+			DNSServiceErrorType err = DNSServiceProcessResult(serviceref);
+			if (err != kDNSServiceErr_NoError)
+				throw Zeroconf::Error(FormatableString("DNSServiceProcessResult: error %0").arg(err));
+//			else
+//				pending.erase(serviceref);
 		}
 	}
 
 	//! Register and announce a target described by an existing Dashel stream
+	Zeroconf::Target::Target(const std::string & name, const int & port) :
+	name(name),
+	port(port)
+	{}
+
+	//! Register and announce a target described by an existing Dashel stream
 	Zeroconf::Target::Target(const Dashel::Stream* stream)
 	{
 		port = atoi(stream->getTargetParameter("port").c_str());
+		name = "Aseba Local " + stream->getTargetParameter("port");
```

Comment:
Is there a reason not to use the copy-constructor for port and name, like `port(atoi(...))` and `name("Aseba Local " + ...)`. But it is a detail.